modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/SkinCleanupTest.java

```diff
@@ -257,6 +260,68 @@ public void testToolBarFocus() {
         assertEquals("first item in toolbar must be focused", bar.getItems().get(0), scene.getFocusOwner());
     }
 
+//-------- TabPane
+    @Test
+    public void testChildrenCountAfterSkinIsReplaced() {
+      TabPane tabPane = new TabPane();
+      tabPane.getTabs().addAll(new Tab("0"), new Tab("1"));
+      showControl(tabPane, false);
+      assertEquals(3, tabPane.getChildrenUnmodifiable().size());
+      TabPaneSkin skin1 = (TabPaneSkin)createAlternativeSkin(tabPane);
+      assertEquals(6, tabPane.getChildrenUnmodifiable().size());
+      tabPane.setSkin(skin1);
+      assertEquals(3, tabPane.getChildrenUnmodifiable().size());
+    }
+
+    @Test
+    public void testChildrenCountAfterSkinIsRemoved() {
+      TabPane tabPane = new TabPane();
+      assertEquals(0, tabPane.getChildrenUnmodifiable().size());
+      tabPane.getTabs().addAll(new Tab("0"), new Tab("1"));
+      showControl(tabPane, false);
+      assertEquals(3, tabPane.getChildrenUnmodifiable().size());
+      root.getChildren().remove(tabPane);
+      assertTrue(tabPane.getSkin() instanceof TabPaneSkin);
+      assertEquals(3, tabPane.getChildrenUnmodifiable().size());
+      tabPane.setSkin(null);
+      assertNull(tabPane.getSkin());
+      assertEquals(0, tabPane.getChildrenUnmodifiable().size());
+    }
```

Comment:
Same as above: what's the reason to actually show?

And just curious:  why is this passing already before the fix?

Reply:
Corrected to use installDefaultSkin().
It works without this fix is because the children are cleared([here](https://github.com/openjdk/jfx/blob/e74f679fda6f03ee8449836147815fdaafb5d626/modules/javafx.controls/src/main/java/javafx/scene/control/Control.java#L300)) in Control class when the skin is set to null.

Reply:
thanks for the info :)